src/background/services/onboarding/handlers/ledgerOnboardingHandler.ts

```diff
@@ -92,20 +101,12 @@ export class LedgerOnboardingHandler implements HandlerType {
       };
     }
 
-    if (xpub) {
+    for (let i = 0; i < (numberOfAccountsToCreate || 1); i++) {
```

Comment:
You should check for `pubKeys.length` as well otherwise it might attempt to add an account we don't have the public key for